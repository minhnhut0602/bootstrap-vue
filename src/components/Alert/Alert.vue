<template>
    <div
        class="alert"
        :class="classes"
    >
        <slot />
    </div>
</template>

<script lang="ts">
import useClasses from '../../composables/useClasses';
import useVariant from '../../composables/useVariant';
import {variantProps} from '../../composables/useVariant';
</script>

<script lang="ts" setup>
const props = defineProps(variantProps);

const {classes} = useClasses([
    useVariant(props.variant, 'alert-{0}').variantClass.value,
]);
</script>
