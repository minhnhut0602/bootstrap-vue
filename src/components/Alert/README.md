# Alert

## Examples

```html
<Alert variant="danger">
    Something went wrong!
</Alert>
```

## Props

| name | type |
| - | - |
| variant | Variant (string) |

## Events

| name | type |
| - | - |
| close | |
| closed | |
