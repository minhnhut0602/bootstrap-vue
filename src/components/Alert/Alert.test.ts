import Alert from '.';

describe('template', () => {
    componentRenderTest(Alert);

    componentSlotRenderTest(Alert);

    componentWrapperClassTest(Alert, {variant: 'danger'}, 'alert-danger');
});
