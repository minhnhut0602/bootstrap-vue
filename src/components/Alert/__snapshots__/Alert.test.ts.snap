// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<div class=\\"alert alert-primary\\">
  <!-- @slot Displays content inside the alert -->
</div>"
`;
